### PR TITLE
DT-6548: Wait card

### PR DESCRIPTION
**Pull request by Antiik91**
## Proposed Changes

  - Refactored wait car to match current design
  - Refactored how the top position of navistack is determined. Calculate it instead of using ever growing number of classes.
  - Refactored the card style to match current desing
  - Switched navi-expand icon to match current desing.

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/component/itinerary/navigator/NaviCard.js [hunk 1]
```diff
@@ -20,16 +21,34 @@ const iconMap = {
...
+    );
+  }
+  if (legType === LEGTYPE.END) {
+    return <FormattedMessage id="navigation-journey-end" />;
+  }
   const iconName = legType === LEGTYPE.WAIT ? iconMap.WAIT : iconMap[leg.mode];
-  let instructions = `navileg-${leg.mode.toLowerCase()}`;
+
+  let instructions = `navileg-${leg?.mode.toLowerCase()}`;
   if (legType === LEGTYPE.TRANSIT) {
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviCard.js]: This logic leaks. WAIT leg gets id navileg-undefined. Maybe harmless, but these if conditions are bit hard to understand. Please rearrange.

File: app/component/itinerary/navigator/NaviCardContainer.js [hunk 2]
```diff
@@ -143,18 +157,13 @@ function NaviCardContainer(
 
   const { first, last } = getFirstLastLegs(realTimeLegs);
   let legType;
-  let naviTopContent;
+  const t = currentLeg ? legTime(currentLeg.start) : time;
+  const nextLeg = realTimeLegs.find(leg => {
+    return legTime(leg.start) > t;
+  });
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviCardContainer.js]: Really a minor thing, but next leg search could be little func just like getFirstLastLegs(). It is done twice in this file.

File: app/component/itinerary/navigator/NaviCardExtension.js [hunk 3]
```diff
@@ -43,12 +50,12 @@ const NaviCardExtension = ({ legType, leg }, { config }) => {
           <RouteNumberContainer
             className={cx('line', vehicleMode.toLowerCase())}
             route={leg.route}
-            mode={leg.mode.toLowerCase()}
+            mode={vehicleMode.toLowerCase()}
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviCardExtension.js]: Why this change? at least HSL stops have sometimes wrong mode data.

File: app/component/itinerary/navigator/NaviCardExtension.js [hunk 4]
```diff
@@ -12,23 +12,30 @@ import { getDestinationProperties, LEGTYPE } from './NaviUtils';
...
   } else {
     destination.iconId = 'icon-icon_mapMarker-to';
     destination.className = 'place';
     destination.name = place;
   }
 
   if (legType === LEGTYPE.TRANSIT) {
-    const { intermediatePlaces } = leg;
+    const { intermediatePlaces, headsign, trip } = leg;
+    const hs = headsign || trip?.tripHeadsign;
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviCardExtension.js]: Transit legs always include trip, no need for ?

File: app/component/itinerary/navigator/NaviInstructions.js [hunk 5]
```diff
@@ -13,7 +13,9 @@ export default function NaviInstructions(
   { leg, nextLeg, instructions, legType },
   { intl, config },
 ) {
-  const { distance, duration } = leg;
+  if (!leg && !nextLeg) {
+    return null;
+  }
   const [fadeOut, setFadeOut] = useState(false);
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviInstructions.js]: incorrect proptypes require both legs

File: app/component/itinerary/navigator/NaviInstructions.js [hunk 6]
```diff
@@ -43,33 +46,50 @@ export default function NaviInstructions(
...
   }
-
-  if (legType === LEGTYPE.WAIT) {
-    const { mode, headsign, route } = nextLeg;
-
-    const color = route.color ? route.color : 'currentColor';
+  if (legType === LEGTYPE.WAIT && nextLeg.mode !== 'WALK') {
+    const { mode, headsign, route, end, start } = nextLeg;
+    const hs = headsign || nextLeg.trip?.tripHeadsign;
+    const color = route?.color ? route.color : 'currentColor';
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/NaviInstructions.js]: route.color || 'currentColor'

**Review by vesameskanen — CHANGES_REQUESTED** [review]
I found only some minor issues that could be changed.

File: app/component/itinerary/navigator/NaviCardExtension.js (hunk 3 above)
Comment (Antiik91) [reply app/component/itinerary/navigator/NaviCardExtension.js]: There was an issue with the leg.mode was not shown, I'll revisit on this and see if it's still the case, and will fix.

**Review by vesameskanen — APPROVED** [review]

**Pull request merged**